addons/portal/static/src/js/portal.js

```diff
@@ -30,12 +30,18 @@ publicWidget.registry.portalDetails = publicWidget.Widget.extend({
      * @private
      */
     _adaptAddressForm: function () {
-        var $country = this.$('select[name="country_id"]');
-        var countryID = ($country.val() || 0);
-        this.$stateOptions.detach();
-        var $displayedState = this.$stateOptions.filter('[data-country_id=' + countryID + ']');
-        var nb = $displayedState.appendTo(this.$state).removeClass('d-none').show().length;
-        this.$state.parent().toggle(nb >= 1);
+        const country = document.querySelector('select[name="country_id"]');
```

Comment:
Hello @risr-odoo, 

I didn't found any significant logical changes in this file. it's look like you have just converted Jquery into Javascript. If so, we should not create extra diff here. Also I review the task specs, the mentioned issue raise only when `account_peppol` is installed so no meaning to fix that issue here. 

I could not find any cause for a traceback in this file(`portal/portal.js`), and the reason you provided in the commit message does not seem particularly relevant to the issue.

`$displayedState.appendTo(this.$state).removeClass('d-none').show().length;` 

As you mentioned show() can only remove the style.display attribute. However, since we already explicitly removed the `d-none` class here, it was already working as expected.

Thanks : )

Reply:
@divy-odoo, Yes when I pushed this commit, few hours ago below commit fixed this bug  `https://github.com/odoo/odoo/commit/7f66c6ba5b1c5b04c0c73b267ef2c33acd6724b1` but when I checked in runbot the issue was not fixed so I did not request to cancel the task but I think in runbot the issue is still persisting because of the `account_peppol` module  is installed.

Reply:
Agree but there is no need to do changes in portal module. You can remove those changes if not neccessary.